pkg/cluster/storage/provision.go

```diff
@@ -0,0 +1,455 @@
+package storage
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"net/url"
+	"reflect"
+	"strings"
+
+	"github.com/goharbor/harbor-cluster-operator/controllers/common"
+	goharborv1 "github.com/goharbor/harbor-operator/apis/goharbor.io/v1alpha2"
+	minio "github.com/goharbor/harbor-operator/pkg/cluster/storage/minio/api/v1"
+	"github.com/goharbor/harbor-operator/pkg/lcm"
+	corev1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1beta1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func (m *MinIOController) ProvisionMinIOProperties(minioInstamnce *minio.Tenant) (*lcm.CRStatus, error) {
+	properties := &lcm.Properties{}
+	data,err := m.getMinIOProperties(minioInstamnce)
+	if err != nil {
+		return minioNotReadyStatus(getMinIOProperties, err.Error()), err
+	}
+	properties.Add(lcm.InClusterSecretForStorage, data)
```

Comment:
I think incluster and external mode will return the same property key, so using `storage` as key directly?

Reply:
right, this name may be confused, I will update it, in actually, LCM-controller not have any logic about external storage.